src/components/select/multi-select/multi-select.test.js

```diff
@@ -0,0 +1,1318 @@
+      selectList().should("be.visible");
+    });
+
+    it("should close the list with mouse click on dropdown button", () => {
+      CypressMountWithProviders(<MultiSelectComponent />);
+
+      dropdownButton().click();
+      dropdownButton().click();
+      selectList().should("not.exist");
+    });
+
+    it("should close the list with the Tab key", () => {
+      CypressMountWithProviders(<MultiSelectComponent />);
+
+      dropdownButton().click();
+      selectList().should("be.visible");
+      selectInput().tab();
+      selectInput().should("have.attr", "aria-expanded", "false");
+      selectList().should("not.exist");
+    });
+
+    it("should close the list with the Esc key", () => {
+      CypressMountWithProviders(<MultiSelectComponent />);
+
+      dropdownButton().click();
+      selectList().should("be.visible");
+      selectInput().trigger("keydown", { ...keyCode("Esc") });
+      selectInput().should("have.attr", "aria-expanded", "false");
+      selectList().should("not.exist");
+    });
+
+    it("should close the list by clicking out of the component", () => {
+      CypressMountWithProviders(<MultiSelectComponent />);
+
+      dropdownButton().click();
+      selectList().should("be.visible");
+      body().click({ force: true });
+      selectInput().should("have.attr", "aria-expanded", "false");
+      selectList().should("not.exist");
+    });
+
+    it.each([
+      ["open", "downarrow"],
+      ["open", "uparrow"],
+      ["open", "Home"],
+      ["open", "End"],
+      ["not open", "Enter"],
+    ])(
+      "should %s the list when %s is pressed with Multi Select input in focus",
+      (state, key) => {
+        CypressMountWithProviders(<MultiSelectComponent />);
+
+        commonDataElementInputPreview().focus();
+        selectInput().trigger("keydown", { ...keyCode(key), force: true });
+        if (state === "open") {
+          selectList().should("be.visible");
+        } else {
+          selectList().should("not.exist");
+        }
+      }
+    );
+
+    it.each([["Amber"], ["Yellow"]])(
+      "should select option %s when clicked from the list and create option pill in the input",
+      (option) => {
+        CypressMountWithProviders(<MultiSelectComponent />);
+
+        dropdownButton().click();
+        selectListText(option).click();
+        selectInput().should("have.attr", "aria-expanded", "true");
+        selectList().should("be.visible");
+        multiSelectPill().should("have.attr", "title", option);
+      }
+    );
+
+    it("should select two options and create option pills in the input", () => {
+      CypressMountWithProviders(<MultiSelectComponent />);
+
+      dropdownButton().click();
+      selectListText(option1).click();
+      selectInput().should("have.attr", "aria-expanded", "true");
+      selectList().should("be.visible");
+      multiSelectPill().should("have.attr", "title", option1);
+      selectListText(option2).click();
+
+      for (let i = 0; i < 2; i++) {
```

Comment:
why do we need to use here a `for` loop and an `if` statement? we can just delete `for` and `if` and use

```suggestion
      multiSelectPillByPosition(0).should("have.attr", "title", option1);
      multiSelectPillByPosition(1).should("have.attr", "title", option2);
```

and that's it :)

Reply:
I did it that way to avoid repeating code for each option.